File: go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	if len(a) == len(b) {

		var errorCount int

		for i := 0; i < len(a); i++ {

			if a[i] != b[i] {
				errorCount++
			}

		}

		return errorCount, nil

	}

	return 0, errors.New("sequences are not equal")
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    if string.len(a) == string.len(b) then
        local error_count = 0

        for i = 1, #a do
            if a:sub(i, i) ~= b:sub(i, i) then
                error_count = error_count + 1
            end
        end

        return error_count
    end
    return -1
end

return Hamming
